### PR TITLE
batch norm tests and fixes

### PR DESCRIPTION
**Pull request by eaidova**

### CODE REVIEW

File: src/frontends/pytorch/src/op/batch_norm.cpp [hunk 1]
```diff
@@ -15,12 +15,29 @@ OutputVector translate_batch_norm(NodeContext& context) {
...
+    auto batch_dim_exp = context.mark_node(std::make_shared<opset8::Unsqueeze>(batch_dim, zero_i));
+    Output<Node> weight = context.mark_node(std::make_shared<opset8::Broadcast>(one_f, channel_dim_exp));
+    Output<Node> bias = context.mark_node(std::make_shared<opset8::Broadcast>(zero_f, channel_dim_exp));
+    if (!context.input_is_none(1)){ 
+        weight = context.get_input(1);
+    }
+    if (!context.input_is_none(2)){
+        bias = context.get_input(2);
+    }
+    // index 3 running_mean and index 4 running_var can be none for training case only, chack that not training before
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/batch_norm.cpp]:
```suggestion
    // index 3 running_mean and index 4 running_var can be none for training case only, check that not training before
```

File: src/frontends/pytorch/src/op/batch_norm.cpp [hunk 2]
```diff
@@ -15,12 +15,29 @@ OutputVector translate_batch_norm(NodeContext& context) {
...
+    auto one_f = context.mark_node(opset8::Constant::create(element::f32, Shape{}, {1}));
+    auto zero_f = context.mark_node(opset8::Constant::create(element::f32, Shape{}, {0}));
+    auto one_i = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {1}));
+    auto minus_one_i = context.mark_node(opset8::Constant::create(element::i64, Shape{1}, {-1}));
+    auto zero_i = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {0}));
+    auto channel_dim = context.mark_node(std::make_shared<opset8::Gather>(input_shape, one_i, zero_i));
+    auto channel_dim_exp = context.mark_node(std::make_shared<opset8::Unsqueeze>(channel_dim, zero_i));
+    auto batch_dim = context.mark_node(std::make_shared<opset8::Gather>(input_shape, zero_i, zero_i));
+    auto batch_dim_exp = context.mark_node(std::make_shared<opset8::Unsqueeze>(batch_dim, zero_i));
+    Output<Node> weight = context.mark_node(std::make_shared<opset8::Broadcast>(one_f, channel_dim_exp));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/batch_norm.cpp]: This is only needed if `weight` is None

File: src/frontends/pytorch/src/op/batch_norm.cpp [hunk 3]
```diff
@@ -15,12 +15,29 @@ OutputVector translate_batch_norm(NodeContext& context) {
...
+    Output<Node> bias = context.mark_node(std::make_shared<opset8::Broadcast>(zero_f, channel_dim_exp));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/batch_norm.cpp]: Same here

**Comment by eaidova** [comment]
@mvafin  all your comments are applied, please review

File: src/frontends/pytorch/src/op/batch_norm.cpp [hunk 4]
```diff
@@ -15,12 +15,36 @@ OutputVector translate_batch_norm(NodeContext& context) {
...
-    auto running_mean = context.get_input(3);
-    auto running_var = context.get_input(4);
+    auto input_shape = context.mark_node(std::make_shared<opset8::ShapeOf>(input));
+    Output<Node> weight;
+    Output<Node> bias;
+    if (!context.input_is_none(1)){ 
+        weight = context.get_input(1);
+    }
+    else {
+        auto zero_i = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {0}));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/batch_norm.cpp]: Can do this broadcasting as function, to reduce code duplication.

**Review by mvafin — APPROVED** [review]

File: src/frontends/pytorch/src/op/batch_norm.cpp (hunk 4 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/batch_norm.cpp]: added
